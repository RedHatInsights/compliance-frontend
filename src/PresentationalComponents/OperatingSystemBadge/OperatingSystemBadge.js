import React from 'react';
import propTypes from 'prop-types';
import { Label } from '@patternfly/react-core';

const OperatingSystemBadge = ({ majorOsVersion }) => {
    const colorMap = {
        default: 'var(--pf-global--disabled-color--200)',
        7: 'cyan',
        8: 'purple'
    };
    const color = colorMap[majorOsVersion] || colorMap.default;

    return <Label { ...{ color } }>
        RHEL { majorOsVersion }
    </Label>;
};

OperatingSystemBadge.propTypes = {
    majorOsVersion: propTypes.string
};

export default OperatingSystemBadge;
