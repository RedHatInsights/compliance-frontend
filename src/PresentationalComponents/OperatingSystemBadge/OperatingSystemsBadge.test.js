import OperatingSystemBadge from './OperatingSystemBadge';

describe('OperatingSystemBadge', () => {
    it('expect to render without error', () => {
        const wrapper = shallow(
            <OperatingSystemBadge />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('expect to render with majorOsVersion given', () => {
        const wrapper = shallow(
            <OperatingSystemBadge majorOsVersion="7" />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
