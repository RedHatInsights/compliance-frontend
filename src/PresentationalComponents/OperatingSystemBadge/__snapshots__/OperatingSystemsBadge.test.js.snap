// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperatingSystemBadge expect to render with majorOsVersion given 1`] = `
<Label
  color="cyan"
>
  RHEL 
  7
</Label>
`;

exports[`OperatingSystemBadge expect to render without error 1`] = `
<Label
  color="var(--pf-global--disabled-color--200)"
>
  RHEL 
</Label>
`;
